Tribler/community/market/core/transaction.py

```diff
@@ -0,0 +1,503 @@
+        """
+        Return the status of this transaction, can be one of these: "pending", "completed", "error".
+        :rtype: str
+        """
+        if len([payment for payment in self.payments if not payment.success]):
+            return "error"
+        return "completed" if self.is_payment_complete() else "pending"
+
+    @staticmethod
+    def unitize(amount, min_unit):
+        """
+        Return an a amount that is a multiple of min_unit.
+        """
+        if Decimal(str(amount)) % Decimal(str(min_unit)) == Decimal(0):
+            return amount
+
+        return (int(amount / min_unit) + 1) * min_unit
+
+    def add_payment(self, payment):
+        self._logger.debug("Adding price %s and quantity %s to transaction %s",
+                           payment.transferee_price, payment.transferee_quantity, self.transaction_id)
+        self._transferred_quantity += payment.transferee_quantity
+        self._transferred_price += payment.transferee_price
+        self._logger.debug("Transferred price: %s, transferred quantity: %s",
+                           self.transferred_price, self.transferred_quantity)
+        self._payments.append(payment)
+
+    def last_payment(self, is_ask):
+        for payment in reversed(self._payments):
+            if is_ask and float(payment.transferee_quantity) > 0:
+                return payment
+            elif not is_ask and float(payment.transferee_price) > 0:
+                return payment
+        return None
+
+    def next_payment(self, order_is_ask, min_unit, incremental):
+        if not incremental:
+            # Don't use incremental payments, just return the full amount
+            ret_val = self.total_quantity if order_is_ask else self.total_price
+            self._logger.debug("Returning %s for the next payment (no incremental payments)", ret_val)
+            return ret_val
+
+        last_payment = self.last_payment(not order_is_ask)
+        if not last_payment:
+            # Just return the lowest unit possible
+            return Quantity(min_unit, self.total_quantity.wallet_id) if order_is_ask else \
+                Price(min_unit, self.total_price.wallet_id)
+
+        # We determine the percentage of the last payment of the total amount
+        if order_is_ask:
+            if self.transferred_price >= self.total_price:  # Complete the trade
+                return self.total_quantity - self.transferred_quantity
+
+            percentage = float(last_payment.transferee_price) / float(self.total_price)
+            transfer_amount = Transaction.unitize(float(percentage * float(self.total_quantity)), min_unit) * 2
+            if transfer_amount > float(self.total_quantity - self.transferred_quantity):
+                transfer_amount = float(self.total_quantity - self.transferred_quantity)
+            return Quantity(transfer_amount, self.total_quantity.wallet_id)
+        else:
+            if self.transferred_quantity >= self.total_quantity:  # Complete the trade
```

Comment:
This is duplicated code please unify these two blocks into a single reused function.